src/board_builder/structures/charuco_board.py

```diff
@@ -0,0 +1,33 @@
+class CharucoBoard:
+    """
+    Representation of the Charuco Board
+    Marker 0 is at the top left, origin is at the bottom left.
+    Positive x goes from left to right, positive y goes upwards, positive z goes out of the board
+    """
+
+    _board_marker_ids = list[int]
+    _board_marker_positions = list[list[int]]
+
+    def __init__(self):
+        self._board_marker_ids = []
+        self._board_marker_positions = []
+
+    def _generate_board(self):
+        for i in range(40):
+            self._board_marker_ids.append(i)
+
+        for marker_id in self._board_marker_ids:
+            x_coords = [30.0, 70.0, 110.0, 150.0, 10.0, 50.0, 90.0, 130.0]
+            x = x_coords[marker_id % 8]
+            y = 190.0 - (marker_id // 4) * 20.0
+            z = 0.0
+            self._board_marker_positions.append([x, y, z])
+
+    def get_ids(self):
+        self._generate_board()
```

Comment:
won't this get called every time you try to get ids or positions? Wouldn't it make more sense for this to be called from __init__?

Reply:
Just to elaborate, from what I see in _generate_board(), it looks like the id's and positions will have their elements added each call. So first call they will have size 40 (ok), but then second it will be 80, then third call will be 120, etc...

Reply:
Yes it does make more sense to put that in init. I will make those changes in my board_builder PR which I will do Monday, so feel free to close this one.

Reply:
Or Tuesday actually